internal/pkg/otel/manager/runtime.go

```diff
@@ -0,0 +1,198 @@
+	}
+
+	if collectorPath == "" {
+		// collector path is required
+		return nil, errors.New("no collector path provided")
+	}
+
+	if _, err := os.Stat(collectorPath); err != nil {
+		// we cannot access the collector path
+		return nil, fmt.Errorf("cannot access collector path: %w", err)
+	}
+
+	httpHealthCheckPort, err := findRandomTCPPort()
+	if err != nil {
+		return nil, fmt.Errorf("could not find port for http health check: %w", err)
+	}
+
+	if err := injectHeathCheckV2Extension(cfg, httpHealthCheckPort); err != nil {
+		return nil, fmt.Errorf("failed to inject health check extension: %w", err)
+	}
+
+	confMap := cfg.ToStringMap()
+	confBytes, err := yaml.Marshal(confMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal config to yaml: %w", err)
+	}
+
+	stdOut := runtime.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+	// info level for stdErr because by default collector writes to stderr
+	stdErr := runtime.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+
+	procCtx, procCtxCancel := context.WithCancel(ctx)
+	processInfo, err := process.Start(collectorPath,
+		process.WithArgs(collectorArgs),
+		process.WithContext(procCtx),
+		process.WithEnv(os.Environ()),
+		process.WithCmdOptions(func(c *exec.Cmd) error {
+			c.Stdin = bytes.NewReader(confBytes)
+			c.Stdout = stdOut
+			c.Stderr = stdErr
+			return nil
+		}),
+	)
+	if err != nil {
+		// we failed to start the process
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: %w", err)
+	}
+	if processInfo.Process == nil {
+		// this should not happen but just in case
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: process is nil")
+	}
+
+	ctl := &procHandle{
+		processDoneCh: make(chan struct{}),
+		processInfo:   processInfo,
+	}
+
+	healthCheckDone := make(chan struct{})
+	go func() {
+		defer func() {
+			close(healthCheckDone)
+		}()
+		currentStatus := componentstatus.StatusStarting
+		reportStatus(ctx, statusCh, aggregateStatus(currentStatus, nil))
+		// we will check the health of the collector every 1 second for the first 10 attempts
+		// until we get a successful response.
+		maxFailedAttempts := 10
+		timerDuration := 1 * time.Second
+		healthCheckTicker := time.NewTimer(timerDuration)
+		currentFailedAttempts := 0
+		defer healthCheckTicker.Stop()
+		for {
+			statuses, err := AllComponentsStatuses(procCtx, httpHealthCheckPort)
+			if err != nil {
+				switch {
+				case errors.Is(err, context.Canceled):
+					reportStatus(ctx, statusCh, aggregateStatus(componentstatus.StatusStopped, nil))
+					return
+				case currentFailedAttempts > maxFailedAttempts:
+					reportStatus(procCtx, statusCh, aggregateStatus(componentstatus.StatusFatalError, err))
+					return
+				}
+				currentFailedAttempts++
+			} else {
+				if reportedStatus := statuses.Status(); currentStatus != reportedStatus {
+					currentStatus = reportedStatus
+					reportStatus(procCtx, statusCh, statuses)
+				}
```

Comment:
It looks like this will only emit the statuses when the overall collector status changes, which I don't think is correct. We want to emit it every time any of the substatuses changes as well, so I think we need to retain the whole struct and do the equivalent of a reflect.DeepEqual on it.

Reply:
ok my original thinking was that this aggregate status will change if an extension/pipeline transitions from OK, to any Error statuses, and vice-versa, which I still believe is correct. That said, I am not quite sure if you meant the following case with your comment, but we might get twice an aggregate status of `StatusRecoverableError` as an example but due to a different components. So yes I will make this always emitting thanks for the catch

Reply:
Always emitting them is also not ideal, because it forces the coordinator loop to process them every time. What I was worried about were less changes from OK to an error status, but rather changes where a component hasn't even started. I'm not sure how healthcheckv2 calculates status in that case, and I'd feel more confident if we assumed as little as possible here.

Reply:
What you say makes sense, but it feels that it should be part of the OTel status manipulation/transformation which IMO should live "closer" to the manager and not in the coordinator. Either way here is the emitting on an actual change aaad4f26ade15ed415c4c85f5c03b115c42e2ec4

Reply:
I think it should live in the otel manager. The fact that the otel manager fetches the status periodically vs subscribing to a GRPC stream of changes is an implementation detail that its callers should not care about.